plugins/become/ksusu.py

```diff
@@ -0,0 +1,182 @@
+              - section: ksusu_become_plugin
+                key: executable_su
+            vars:
+              - name: ansible_become_exe_su
+              - name: ansible_ksusu_exe_su
+            env:
+              - name: ANSIBLE_BECOME_EXE
+              - name: ANSIBLE_BECOME_EXE_SU
+              - name: ANSIBLE_KSUSU_EXE_SU
+        become_flags_ksu:
+            description: Options to pass to C(ksu).
+            default: '-Z -q'
+            ini:
+              - section: privilege_escalation
+                key: become_flags_ksu
+              - section: ksusu_become_plugin
+                key: flags_ksu
+            vars:
+              - name: ansible_become_flags_ksu
+              - name: ansible_ksusu_flags_ksu
+            env:
+              - name: ANSIBLE_BECOME_FLAGS_KSU
+              - name: ANSIBLE_KSU_FLAGS_KSU
+        become_flags:
+            description: Options to pass to C(su).
+            default: ''
+            ini:
+              - section: privilege_escalation
+                key: become_flags
+              - section: privilege_escalation
+                key: become_flags_su
+              - section: ksusu_become_plugin
+                key: flags_su
+            vars:
+              - name: ansible_become_flags
+              - name: ansible_become_flags_su
+              - name: ansible_ksusu_flags
+              - name: ansible_ksusu_flags_su
+            env:
+              - name: ANSIBLE_BECOME_FLAGS
+              - name: ANSIBLE_BECOME_FLAGS_SU
+              - name: ANSIBLE_KSU_FLAGS_SU
+        become_pass:
+            description: C(ksu) password.
+            required: False
+            vars:
+              - name: ansible_ksusu_pass
+              - name: ansible_become_pass
+              - name: ansible_become_password
+            env:
+              - name: ANSIBLE_BECOME_PASS
+              - name: ANSIBLE_KSUSU_PASS
+            ini:
+              - section: ksusu_become_plugin
+                key: password
+        prompt_l10n:
+            description:
+                - List of localized strings to match for prompt detection.
+            default: ['Kerberos password for .*@.*:']
+            ini:
+              - section: ksusu_become_plugin
+                key: localized_prompts
+            vars:
+              - name: ansible_ksusu_prompt_l10n
+            env:
+              - name: ANSIBLE_KSUSU_PROMPT_L10N
+"""
+
+import re
+
+from ansible.module_utils.common.text.converters import to_bytes
+from ansible.plugins.become import BecomeBase
+
+
+class BecomeModule(BecomeBase):
+
+    name = 'community.general.ksusu'
+
+    # messages for detecting prompted password issues
+    fail = ('Password incorrect',)
+    missing = ('No password given',)
+
+    def check_password_prompt(self, b_output):
+        ''' checks if the expected password prompt exists in b_output '''
+
+        prompts = self.get_option('prompt_l10n') or ["Kerberos password for .*@.*:"]
```

Comment:
In https://github.com/russoz-ansible/community.general/blob/main/plugins/module_utils/cmd_runner.py#L243-L247 we take a different approach to this: we force the execution to use the locale C (by default), that should exit in virtually every system. But your way also works. :-)

Reply:
@russoz that won't work since become plugins have no control over the environment.

Reply:
This is now getting the default in a different manner per what @felixfontein wanted. 

Also note though that this is still exactly what the existing ksu.py does so if it is incorrect here it is incorrect there as well.

The current version of MIT Kerberos uses gettext but only has translation of the prompt string to German ("Kerberos-Passwort für"). I also checked Heimdal but it would appear Heimdal does not have ksu. I'm happy to include this but I don't know how to do this within ansible.

Reply:
It wasn'a matter of being incorrect, only different ways to achieve the goal. But it doesn't really matter because, as pointed out by @felixfontein , plugins have no control over environment variables.